src/constants/products.ts

```diff
@@ -1,3 +1,33 @@
+import { ProductTypes } from '../../generated/product-types';
+
+export const productFilterLabelByType = {
```

Comment:
both constants in this file need to be dynamically generated somehow every time we add a new product type. it's not ideal to have to maintain them manually.

Reply:
Yep, I missed a commit. Fixed now 👍

Reply:
There are some issues though, the labels in this file are different from the types themselves. Also we do not display all of them, so I couldn't just take the generated `ProductTypeName`'s. I would gladly unify this, but it will make the products filter have even more entries

Reply:
the only product types we wouldn't want to show are those that have only "deprecated" products associated with them

Reply:
Hey, as I mentioned here https://github.com/NexusMutual/frontend-next/pull/828#discussion_r1719943296 - I don't understand why these mappings are needed at all - the `productTypes` data structure can be used directly for the filter component. We agreed in DM to move the `ProductTypes` enum from FE app to the sdk to help with TS types, but I don't see the need for the mappings structures at all. Regarding the deprecated ones, we should tweak the build script here to mark them so and that should be it I believe.

Reply:
> Hey, as I mentioned here [NexusMutual/frontend-next#828 (comment)](https://github.com/NexusMutual/frontend-next/pull/828#discussion_r1719943296) - I don't understand why these mappings are needed at all - the `productTypes` data structure can be used directly for the filter component. We agreed in DM to move the `ProductTypes` enum from FE app to the sdk to help with TS types, but I don't see the need for the mappings structures at all. Regarding the deprecated ones, we should tweak the build script here to mark them so and that should be it I believe.

I initially took this approach because the labels are different than the product type names, and not all of them were displayed. Managing these in the sdk seemed easier, but got way to complex way to fast. Since Rox' comment above I used the products from the sdk instead and took your suggested approach. I moved the complexity from the sdk to the frontend, you can check it [here](https://github.com/NexusMutual/frontend-next/pull/828/files#diff-e04183a63119c21b1729554148fe92aaaa8c618f926bbb57173a5a49490018ebR43).

This means the options text will be tied to the productTypes text, and will update automatically 👍

Reply:
I changed this pr to _just_ adding the `ProductTypes` type, since the other stuff is no longer needed anymore. No rush to merge this, since the same constant is available in the frontend-react project